File: scripts/run_migrations.py
#!/usr/bin/env python3
"""
Execute Supabase migrations
Usage: python scripts/run_migrations.py
"""

import os
import glob
import sys
from pathlib import Path

try:
    import psycopg2
    from psycopg2 import sql
except ImportError:
    print("❌ psycopg2 not installed")
    print("   Install with: pip install psycopg2-binary")
    sys.exit(1)

SUPABASE_URL = os.getenv("NEXT_PUBLIC_SUPABASE_URL")
SERVICE_KEY = os.getenv("SUPABASE_SERVICE_ROLE_KEY")

if not SUPABASE_URL or not SERVICE_KEY:
    print("❌ Missing environment variables")
    sys.exit(1)

# Parse Supabase URL
url_parts = SUPABASE_URL.replace("https://", "").split(".supabase.co")
project_id = url_parts[0]
db_url = f"postgresql://postgres:{SERVICE_KEY}@{project_id}.supabase.co:5432/postgres"

print(f"🔗 Connecting to Supabase: {project_id}")

try:
    conn = psycopg2.connect(db_url)
    conn.autocommit = True
    cur = conn.cursor()
    print("✅ Connected\n")

    # Find migrations
    migrations_dir = Path(__file__).parent.parent / "supabase" / "migrations"
    migration_files = sorted(glob.glob(str(migrations_dir / "076_*.sql")))
    migration_files.extend(sorted(glob.glob(str(migrations_dir / "077_*.sql"))))
    migration_files.extend(sorted(glob.glob(str(migrations_dir / "078_*.sql"))))
    migration_files.extend(sorted(glob.glob(str(migrations_dir / "079_*.sql"))))
    migration_files.extend(sorted(glob.glob(str(migrations_dir / "080_*.sql"))))
    migration_files.extend(sorted(glob.glob(str(migrations_dir / "081_*.sql"))))

    completed = 0
    failed = 0

    for migration_file in migration_files:
        try:
            filename = Path(migration_file).name
            print(f"📝 Running: {filename}")

            with open(migration_file, "r") as f:
                sql_content = f.read()

            # Execute each statement
            statements = [s.strip() for s in sql_content.split(";") if s.strip()]
            for stmt in statements:
                cur.execute(stmt)

            print(f"✅ {filename}\n")
            completed += 1

        except Exception as e:
            print(f"❌ {filename}")
            print(f"   Error: {str(e)}\n")
            failed += 1

    print("=" * 50)
    print(f"📊 Results: {completed} completed, {failed} failed")

    if failed > 0:
        sys.exit(1)

    print("✨ All migrations completed!")

except Exception as e:
    print(f"❌ Connection error: {str(e)}")
    sys.exit(1)

finally:
    try:
        cur.close()
        conn.close()
    except:
        pass

File: scripts/run-migrations.ts
#!/usr/bin/env node
/**
 * Run Supabase migrations
 * Usage: npx ts-node scripts/run-migrations.ts
 * Requires: NEXT_PUBLIC_SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY
 */

import { createClient } from "@supabase/supabase-js";
import * as fs from "fs";
import * as path from "path";

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error("❌ Missing environment variables:");
  console.error("   - NEXT_PUBLIC_SUPABASE_URL");
  console.error("   - SUPABASE_SERVICE_ROLE_KEY");
  process.exit(1);
}

const client = createClient(supabaseUrl, supabaseKey);

async function runMigrations() {
  console.log("🚀 Starting Supabase migrations...\n");

  const migrationsDir = path.join(__dirname, "../supabase/migrations");
  const files = fs
    .readdirSync(migrationsDir)
    .filter((f) => f.endsWith(".sql") && f >= "076_") // Only Phase 1 and later
    .sort();

  let completed = 0;
  let failed = 0;

  for (const file of files) {
    try {
      const sql = fs.readFileSync(path.join(migrationsDir, file), "utf-8");

      console.log(`📝 Running: ${file}`);

      // Split by ; to handle multiple statements
      const statements = sql.split(";").filter((s) => s.trim());

      for (const statement of statements) {
        const trimmed = statement.trim();
        if (!trimmed) continue;

        let error: { message?: string } | null = null;
        try {
          const res = await client.rpc("exec", { query: trimmed });
          error = res.error;
        } catch {
          // Fallback: use raw query
          const res = await client.from("_migrations").select().limit(0);
          error = res.error;
        }

        if (error && error.message !== "No rows found") {
          throw error;
        }
      }

      console.log(`✅ ${file}`);
      completed++;
    } catch (error) {
      console.error(`❌ ${file}`);
      console.error(`   Error: ${(error as Error).message}\n`);
      failed++;
    }
  }

  console.log("\n" + "=".repeat(50));
  console.log(`📊 Results: ${completed} completed, ${failed} failed`);

  if (failed > 0) {
    process.exit(1);
  }

  console.log("✨ All migrations completed successfully!");
}

runMigrations();
